toolchain/check/handle_class.cpp

```diff
@@ -512,6 +512,15 @@ auto HandleParseNode(Context& context, Parse::BaseDeclId node_id) -> bool {
     return true;
   }
 
+  if (!context.struct_type_fields_stack().PeekArray().empty()) {
+    CARBON_DIAGNOSTIC(BaseDeclAfterFieldDecl, Error,
+                      "`{0}` declaration appears after field declaration(s)",
+                      Lex::TokenKind);
+    context.emitter().Emit(node_id, BaseDeclAfterFieldDecl,
+                           Lex::TokenKind::Base);
```

Comment:
Why do you pass in `Base` here, versus putting `base` directly in the message?

Reply:
Inspired by some nearby code (`DiagnoseClassSpecificDeclRepeated` - which admittedly is actually parameterized, with one caller using `Lex::TokenKind::Base` and another using `Lex::TokenKind::Adapt`, whereas this case only has the one value). Smallest argument in favor of type safety/write-it-once - there's one bit of code for mapping this keyword to text, rather than including freeform text that is consistent today, but isn't checked in any way... 
But I certainly don't feel strongly. I see there are at least several cases where it's written literally/in the string, so I'll change to that.